### PR TITLE
Reduce Try/TryFrom implementations for general types

### PR DESCRIPTION
**Pull request by squadgazzz**
# Description
During some debugging sessions, it was noticed that having implicit conversions overcomplicate code navigations. The proposal is to use explicit `from/into_domain` functions for general types.

# Changes

The PR touches only Auction and FeePolicy structs at the moment. But once bumped into the same issue with any other struct, a follow-up should be opened.

### CODE REVIEW

**Comment by squadgazzz** [comment]
> Why don't we do it for all in one sitting? 🤔

The issue is with general structs only. All the internal fields can still use those implicit conversations.

**Review by mstrug — APPROVED** [review]
Looks ok.

File: crates/autopilot/src/infra/persistence/dto/order.rs [hunk 1]
```diff
@@ -353,13 +323,49 @@ impl From<FeePolicy> for domain::fee::Policy {
...
+            order_uid,
+            sell_amount: self.sell_amount.into(),
+            buy_amount: self.buy_amount.into(),
+            fee: self.fee.into(),
+            solver: self.solver.into(),
+        }
+    }
+}
+
+impl From<domain::Quote> for Quote {
```
[the remark this answers is not in the document]
Comment (squadgazzz) [reply crates/autopilot/src/infra/persistence/dto/order.rs:358]: Please elaborate.

[the remark this answers is not in the document]
Comment (squadgazzz) [reply crates/autopilot/src/infra/persistence/dto/order.rs]: Ah, true, missed that for some reason. Fixed.

**Pull request merged**